### DISCUSSION TITLE
Can't see Unifi devices names (Private reverse DNS)

### DISCUSSION THREAD

**Discussion by Gitonthegrid** [AdguardTeam/AdGuardHome#6819] [General]
Hi all,

For some reason I can't get the **Private reverse DNS** to work. From what I know, I should get the Unifi device names within Adguard if I add the following to the **Private reverse DNS server** setting: (192.168.0.1 is my Unifi router)

```
192.168.0.1
[/168.192.in-addr.arpa/]192.168.0.1
```

I have waited for 24 hours now and rebooted my Adguard, router and some devices, but I don't see the device names specified in Unifi. Under cliënt settings I do see **some** devices, but not all. And most of them don't have the name I have given within Unifi.

Am I doing something wrong here?
I don't know if it is important to mention, but I do use Unbound as my upstream server. So the upstream server says:

`127.0.0.1:5353`

**Comment by hagezi** [comment]
@Gitonthegrid 

Add `[/168.192.in-addr.arpa/]192.168.0.1` to `Settings > DNS settings > Upstream DNS servers`
Add `192.168.0.1` to `Settings > DNS settings > Private reverse DNS servers`

Restart AdGuard Home.

**Reply by Gitonthegrid** [reply]
And can I force a device to get the new name from Unifi?

I changed the settings and did reboot Adguard, but still nog good names.

And what about VLAN devices, will they be recognised as well with these settings?

**Reply by hagezi** [reply]
> And can I force a device to get the new name from Unifi?

This should be done cyclically, or at the latest after a restart.

> I changed the settings and did reboot Adguard, but still nog good names.

What kind of names are displayed? The DHCP server must be running on the Unifi.
Try `[///0.168.192.in-addr.arpa/]192.168.0.1`

> And what about VLAN devices, will they be recognised as well with these settings?

`[///0.168.192.in-addr.arpa/]192.168.0.1` forwards all reverse requests for 192.168.0.* to 192.168.0.1 If you have other segments, you must configure them in the same way.

**Reply by Gitonthegrid** [reply]
for example, 

**I have this device in Unifi "Marantz"**
![image](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/c7c4a6cd-4b05-4be3-ab1c-c99c4726d169)


**But in Adguard it shows the name as 00067831592B:**
![image](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/581373c2-1be6-4896-8819-0ae7c1fde5db)

Good to know about the VLAN section, I have VLANs setup as 192.168.100.1 etc. So this should be good. 
Wat is de difference with this part: `[///0.168.192.in-addr.arpa/]192.168.0.1`?

**Reply by Gitonthegrid** [reply]
For all networks the DHCP is on in Unifi, and I have point the DNS server setting to the Adguard server for all networks in Unifi

**Reply by hagezi** [reply]
Strange, I don't know the Unifi specifics and why AGH displays a random number instead of the name. 

Have a look at the QueryLog there you should actually see the local arpa requests and then you can also see whether they are forwarded to the Unifi. 

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/104343908/8a2c2317-5b39-402b-85b4-3a4be9771265)



I checked my AGH instance and saw that I have three slashes instead of one: `[///178.168.192.in-addr.arpa/fritz.box/]192.168.178.1`

**Reply by Gitonthegrid** [reply]
This is a response from a laptop:

![image](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/b1a3f402-04bb-490b-a56c-912e0dfe98e4)

**Reply by hagezi** [reply]
These are the details of the request of catalog.gamepass.com which of course go to the Unbound. You have to look at the details of a local arpa request.

**Reply by Gitonthegrid** [reply]
These are some request if I search on "arpa" in the query log

![165](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/13188c13-5f42-4eb6-a045-e6d54d1a1014)
![19](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/a98ca705-f21d-4d5e-92f9-ab1cde8e1243)

**Reply by hagezi** [reply]
This is a discovery DNS request, but the requests end up at the Unifi, so the forwarding works as intended. Unfortunately, I can't tell you why it doesn't display the names correctly.

**Reply by Gitonthegrid** [reply]
Can it be a firewall issue maybe?

And is there any explanation why I miss some client devices in Adguard?
I have more clients in Unifi, but they dont show up in Adguard.

Thanks for all the help so far!

**Reply by hagezi** [reply]
Yes, that may be the case, you need to check your network environment. It is now configured correctly in AGH.

**Reply by Gitonthegrid** [reply]
@hagezi 

Oke I found something. 

In the Unifi if i look at the Marantz, it says as hostname the name that is displayed in Adguard  **00067831592B** 
I thought It was looking at the "friendly name"
![image](https://github.com/AdguardTeam/AdGuardHome/assets/74937678/73c8447b-2ea6-4d59-9080-c245d3de207d)

Is it possible that it looks at the "Friendly name" instead of the "hostname"?

**Reply by hagezi** [reply]
https://community.ui.com/questions/Reverse-DNS-lookups-in-Controller-greater-Clients/f9609d03-3b21-468f-bd8f-42b4a22adc61

**Reply by Gitonthegrid** [reply]
Thanks, but I cant find those settings, there has changed alot in 6 years ;)

At least I am a bit further, I now know that Adguard looks at the hostname and not the friendly name (what I was hoping for)
